src/EnergyPlus/WaterCoils.cc

```diff
@@ -6348,14 +6304,15 @@ void CheckForSensorAndSetPointNode(EnergyPlusData &state,
     // a setpoint is also specified on the water coil outlet node
     if (!NodeNotFound) {
         if (WhichCoil > 0) {
-            if (state.dataWaterCoils->WaterCoil(CoilNum).WaterCoilType == DataPlant::PlantEquipmentType::CoilWaterDetailedFlatCooling) {
+            std::string WaterCoilType; // water coil type
+            if (state.dataWaterCoils->WaterCoil(WhichCoil).WaterCoilType == DataPlant::PlantEquipmentType::CoilWaterDetailedFlatCooling) {
                 WaterCoilType = "Coil:Cooling:Water:DetailedGeometry";
```

Comment:
For stuff like this, there should be a `constexpr std::array<std::string_view, (int)PlantEquipmentType::Num> PlantEquipmentTypeNames;` that you can lookup using the enumeration rather than performing this local mapping using conditionals.